.github/workflows/testing.yml

```diff
@@ -19,7 +25,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: ${{ fromJson(inputs.python-versions) }}
         os: [ubuntu-latest, macos-13, windows-latest]
```

Comment:
I like this, and I think that in neuroconv it makes sense because we do use all the oses as developers (I use Linux, you and Ben use Mac, Cody uses Windows I think that Alessandra uses Windows as well, not sure what Szjona uses).

That said, Windows and Linux are sufficiently different and I would like to catch those errors early. 

A compromise would be to do Linux and Windows latest and then once the PR is ready add arm and intel for Mac. What do you think?

Reply:
> That said, Windows and Linux are sufficiently different and I would like to catch those errors early.

I thought the whole point of paring down the testing matrix while a PR is on draft is bc there are likely to be actual bugs in the code (not related to external factors like os, python version, etc.) that should be resolved first, while the substantive development happens. Then, when the PR is ready for review, we can do a full test matrix to make sure that it passes under all relevant conditions (python version, os, etc).

In other words, if there are intrinsic bugs in the code, they can be caught by one OS and one python version, which saves us a whole ton of runners.

Reply:
I was not thinking exactly like that. Maybe your approach makes more sense but this is how I was thinking:

There must be a Goldilocks zone of testing that detects most of the early errors or you code quickly and without a lot of CI overhead. Some use cases to make it concrete:
1) Environment issues. Your local environment assumes something that is not true. 
2) Your code fixes your problem but breaks the code somewhere else.
3) CI issues that can't be tested locally.
4) The user does not have gin-data or wants to see if the gin-data is propagated to the CI.

Once you have fixed this and you feel more confident about the PR then the full tests in ready-to-review catch the long-tail of the errors if there is anything.

From my experience, there are some shallow errors that a Linux coder can make that don't work on Windows (hard coding paths posix style with `/` being a typical one) and that's why I wanted to have Windows.

But this is not that strong of an argument to me. If you think that Linux only is better we can start with only Linux and then if many simple window errors appear we can add windows in the future. What do you think?

Reply:
> There must be a Goldilocks zone of testing that detects most of the early errors or you code quickly and without a lot of CI overhead.

This framework makes sense, I just think that the best bang for our buck is with 1 OS. The majority of cases you are describing (1-4) should be caught regardless of OS, and Windows + Linux is twice as many tests as Linux-only.  Plus, I really don't see any concrete downside to fixing these os-specific bugs (like hard-coding posix-style paths) later in the development cycle.

Reply:
Those are good points, @pauladkisson. Let's move forward in that direction.

Reply:
OK, I did this in the last commit. Also, now that https://github.com/catalystneuro/neuroconv/pull/1061 is merged, I re-added mac with intel chips as discussed in https://github.com/catalystneuro/neuroconv/pull/1080/files.

If this works I will refactor the repeated lines using env variables and then this should be ready to go.